engine/api/workflow/process_requirements.go

```diff
@@ -36,29 +32,63 @@ func getNodeJobRunRequirements(db gorp.SqlExecutor, j sdk.Job, run *sdk.Workflow
 			errm.Append(errValue)
 			continue
 		}
-		sdk.AddRequirement(&requirements, v.ID, name, v.Type, value)
+
 		if v.Type == sdk.ServiceRequirement {
 			containsService = true
 		}
 		if v.Type == sdk.ModelRequirement {
+			// It is forbidden to have more than one model requirement.
+			if model != "" {
+				errm.Append(sdk.ErrInvalidJobRequirementDuplicateModel)
+				break
+			}
 			model = value
 		}
+
+		sdk.AddRequirement(&requirements, v.ID, name, v.Type, value)
 	}
 
 	var modelType string
 	if model != "" {
+		// Load the worker model
 		wm, err := worker.LoadWorkerModelByName(db, model)
 		if err != nil {
 			log.Error("getNodeJobRunRequirements> error while getting worker model %s: %v", model, err)
+			errm.Append(sdk.ErrNoWorkerModel)
 		} else {
+			// Check that the worker model is in an exec group
+			if !sdk.IsInInt64Array(wm.GroupID, execsGroupIDs) {
+				errm.Append(sdk.ErrInvalidJobRequirementWorkerModelPermission)
+			}
+
+			// Check that the worker model has the binaries capabilitites
+			// only if the worker model doesn't need registration
+			if !wm.NeedRegistration && !wm.CheckRegistration {
+				for _, req := range requirements {
+					if req.Type == sdk.BinaryRequirement {
+						var hasCapa bool
+						for _, cap := range wm.RegisteredCapabilities {
+							if cap.Value == req.Value {
+								hasCapa = true
```

Comment:
if hasCapa == true -> could add a break to avoid range over all RegisteredCapabilities